### DISCUSSION TITLE
Multiple (concurrent) consumers for a single event with resilience

### DISCUSSION THREAD

**Discussion by pascal08** [symfony/symfony#44226] [Q&A]
Let's consider a monorepo with n-loosely coupled services `Service-1`, `Service-2`, ... , `Service-n`. `Service-1` emits an event that is consumed by multiple other consuming services. The processing of each event by the consuming service is atomic (1) and independent (2). 

(1) Either the processing succeeds or fails.
(2) If one service fails to process the event, other consuming services are not affected and it will retry the processing later (resilience).

I'd argue that this scenario is currently not supported by symfony/messenger without custom logic to bind messages to handlers for a specific context. I'd like to know it this is true. I'd also like to know if others have dealt with the same situation and how it was solved. I think it's a fairly common setup and I'm a little bit surprised I couldn't find any support for this. I'd be pleased to help building support for this scenario if there's enough common ground.

**Comment by michaoj** [comment]
I think you can use messenger in your use case : you just need to create a queue per service and dispatch the message in all the desired queue. 

Like this if processing message in `queue_service_1` fails, service 2 can still processes the message and so on

**Reply by pascal08** [reply]
Thank you for your reply. 

How would you prevent messages from `queue_service_1` being handled by a message handler in `Service2`?

**Reply by michaoj** [reply]
I would write an `AbstractHandler` that consume the message with the logic and then extends this class in my handlers.

`HandlerService1 extends AbstractHandler` would listen to message in `queue_service_1`
`HandlerService2 extends AbstractHandler` would listen to message in `queue_service_2`

**Reply by pascal08** [reply]
I fail to see how the `AbstractHandler` knows where to route the message to. Would the message envelope needs a stamp for that?
